common/script/index.coffee

```diff
@@ -72,13 +72,51 @@ api.daysSince = (yesterday, options = {}) ->
   Math.abs api.startOfDay(_.defaults {now:yesterday}, o).diff(api.startOfDay(_.defaults {now:o.now}, o), 'days')
 
 ###
-  Should the user do this taks on this date, given the task's repeat options and user.preferences.dayStart?
+  Should the user do this task on this date, given the task's repeat options and user.preferences.dayStart?
 ###
-api.shouldDo = (day, repeat, options={}) ->
-  return false unless repeat
+api.shouldDo = (day, dailyTask, options = {}) ->
+  return false unless dailyTask.type == 'daily' && dailyTask.repeat
+  # MIGRATION HACK, FIXME
+  if dailyTask.startDate == null
+    console.log("null!!!!!")
+    dailyTask.startDate = moment().toDate();
+  if dailyTask.startDate instanceof String
+    console.log("startDate is a string: " + dailyTask.startDate + ". Converting to date");
+    dailyTask.startDate = moment(dailyTask.startDate).toDate();
   o = sanitizeOptions options
-  selected = repeat[api.dayMapping[api.startOfDay(_.defaults {now:day}, o).day()]]
-  return selected
+  dayOfWeek = api.startOfDay(_.defaults {now:day}, o).day()
+
+  # check if event is in the future
+  hasStartedCheck = moment(day).isAfter(dailyTask.startDate) || moment(day).isSame(dailyTask.startDate)
+
+  if dailyTask.frequency == 'daily'
+    daysSinceTaskStart = api.numDaysApart(day, dailyTask.startDate, o)
+    everyXCheck = (daysSinceTaskStart % dailyTask.everyX == 0)
+    return everyXCheck && hasStartedCheck
+  else if dailyTask.frequency == 'weekly'
+    dayOfWeekCheck = dailyTask.repeat[api.dayMapping[dayOfWeek]];
+    weeksSinceTaskStartWeek = api.numWeeksApart(day, dailyTask.startDate, o)
+    everyXCheck = (weeksSinceTaskStartWeek % dailyTask.everyX == 0)
+    return dayOfWeekCheck && everyXCheck && hasStartedCheck
+  else
+    # unexpected frequency string
+    return true
```

Comment:
When it returns true, it displays as due, right? If it can't be calculated, would we want it to be displayed as due or gray? I'd vote gray, but am open to other opinions on the matter.